airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/migrations/SnowflakeAbMetaAndGenIdMigration.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.snowflake.migrations
+
+import io.airbyte.cdk.db.jdbc.JdbcDatabase
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.destination.jdbc.ColumnDefinition
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition
+import io.airbyte.commons.json.Jsons
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationHandler
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationInitialStatus
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.base.destination.typing_deduping.migrators.Migration
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.LinkedHashMap
+
+private val log = KotlinLogging.logger {}
+
+class SnowflakeAbMetaAndGenIdMigration(private val database: JdbcDatabase) :
+    Migration<SnowflakeState> {
+    override fun migrateIfNecessary(
+        destinationHandler: DestinationHandler<SnowflakeState>,
+        stream: StreamConfig,
+        state: DestinationInitialStatus<SnowflakeState>
+    ): Migration.MigrationResult<SnowflakeState> {
+        if (!state.initialRawTableStatus.rawTableExists) {
+            // The raw table doesn't exist. No migration necessary. Update the state.
+            log.info {
+                "Skipping airbyte_meta/generation_id migration for ${stream.id.originalNamespace}.${stream.id.originalName} because the raw table doesn't exist"
+            }
+            return Migration.MigrationResult(state.destinationState, false)
+        }
+
+        val results =
+            database.queryJsons(
+                "SHOW COLUMNS IN TABLE \"${stream.id.rawNamespace}\".\"${stream.id.rawName}\""
+            )
+        val rawTableDefinition =
+            results
+                .groupBy { it.get("schema_name").asText()!! }
```

Comment:
we're already restricting to specifically airbyte_internal.whatever, why do we need to group by schema name / table name?

Reply:
just to avoid. list has 1 element and `first()` call i guess.

Reply:
I think you get the same result with this?
```kotlin
TableDefinition(
            results.associateTo(LinkedHashMap()) {
                // return value of data_type in show columns is a json string.
                val dataType = Jsons.deserialize(it.get("data_type").asText())
                it.get("column_name").asText()!! to
                    ColumnDefinition(
                        it.get("column_name").asText(),
                        dataType.get("type").asText(),
                        0,
                        dataType.get("nullable").asBoolean(),
                    )
            }
        )
```

i.e. don't need to group on schema/table name at all, just directly build the map of column name to definition

(... but also, I would hope that kotlin has some easy `.first` accessor :P )